src/groups/orthogonal_group.jl

```diff
@@ -0,0 +1,528 @@
+        c[2] = X[1, 3]
+        c[3] = X[2, 1]
+        k = 4
+        for i in 4:n, j in 1:(i - 1)
+            c[k] = X[i, j]
+            k += 1
+        end
+    end
+    return c
+end
+
+_doc_get_vector_On = """
+    get_vector(G::OrthogonalLieAlgebra, e, c, ::DefaultLieAlgebraOrthogonalBasis)
+    get_vector(G::SpecialOrthogonalLieAlgebra, e, c, ::DefaultLieAlgebraOrthogonalBasis)
+    get_vector!(G::OrthogonalLieAlgebra, X, e, c, ::DefaultLieAlgebraOrthogonalBasis)
+    get_vector!(G::SpecialOrthogonalLieAlgebra, X, e, c, ::DefaultLieAlgebraOrthogonalBasis)
+
+Compute the tangent vector ``X ∈ 𝔬(n)`` based on a vector of coordinates ``c ∈ ℝ^d``,
+where ``d`` is the dimension of the Lie algebra of the [`OrthogonalGroup`](@ref) `O(n)`
+and the coordinates are with respect to the [`DefaultLieAlgebraOrthogonalBasis`](@ref).
+This is also the version used in [`hat`](@ref).
+
+For ``O(2)`` there is only one coefficient ````c = (α)^$(_tex(:transp))`` and hence
+``X = $(_tex(:pmatrix, "0 & -α", "α & 0"))`` is returned.
+
+For ``n=3`` a usual representtion turns ``c = (α, β, γ)^$(_tex(:transp)) ∈ ℝ^3`` into
+```math
+    X = $(_tex(:pmatrix, "0 & -γ & β", "γ & 0 & -α", "-β & α & 0")),
+```
+hence the coordinate vector is .
+
+For `n ≥ 4`` all further coefficients are used to fill up the following rows of the lower
+triangular part – which determines the upper triangular part due to skew-symmetry
+"""
+
+@doc "$(_doc_get_vector_On)"
+get_vector(𝔤::OrthogonalLieAlgebra, X, ::DefaultLieAlgebraOrthogonalBasis)
+
+@doc "$(_doc_get_vector_On)"
+get_vector!(𝔤::OrthogonalLieAlgebra, c, X::DefaultLieAlgebraOrthogonalBasis)
+
+function get_vector_lie!(
+    ::CommonUnitarySubAlgebra{ManifoldsBase.ℝ,<:ManifoldsBase.TypeParameter{Tuple{2}}},
+    X,
+    c,
+    ::DefaultLieAlgebraOrthogonalBasis{ℝ},
+)
+    @assert size(X) == (2, 2)
+    @assert size(c) == (1,)
+    X[1, 1] = 0.0
+    X[2, 1] = c[1]
+    X[1, 2] = -c[1]
+    X[2, 2] = 0.0
+    return X
+end
+function get_vector_lie!(
+    𝔤::CommonUnitarySubAlgebra{ℝ,<:ManifoldsBase.TypeParameter{Tuple{n}}},
+    X,
+    c,
+    ::DefaultLieAlgebraOrthogonalBasis{ℝ},
+) where {n}
+    @assert size(X) == (n, n)
+    @assert length(c) == manifold_dimension(𝔤)
+    @assert n > 2
+    _get_vector_lie_On!(X, c)
+    return X
+end
+function get_vector_lie!(
+    𝔤::CommonUnitarySubAlgebra{ManifoldsBase.ℝ}, X, c, ::DefaultLieAlgebraOrthogonalBasis{ℝ}
+)
+    G = 𝔤.manifold
+    n = ManifoldsBase.get_parameter(G.manifold.size)[1]
+    @assert length(c) == manifold_dimension(𝔤)
+    @assert size(X) == (n, n)
+    if n == 2
+        X[1, 1] = 0.0
+        X[2, 1] = c[1]
+        X[1, 2] = -c[1]
+        X[2, 2] = 0.0
+    else
+        _get_vector_lie_On!(X, c)
+    end
+    return X
+end
+
+function _get_vector_lie_On!(X, c)
+    n = size(X, 1)
+    @inbounds begin
+        X[1, 2] = -c[3]
+        X[2, 1] = c[3]
+        X[1, 3] = c[2]
+        X[3, 1] = -c[2]
+        X[2, 3] = -c[1]
+        X[3, 2] = c[1]
```

Comment:
It looks like you removed zeroing the diagonal when copying this from Manifolds.jl? That's most likely the cause of the recent failure on CI but maybe you had an idea behind that?

Reply:
If I had an idea, same as in one of the last posts, I forgot about it by now :D (this PR is really a bit long time wise by now) so if that caused CI to fail it should be re-added, if they make CI fail now.
I actually can not even find where that is in Manifolds.jl...